### ISSUE TITLE
Implement a feature to consume range-like source with a fixed size task pool

### ISSUE THREAD

**Issue by mihails-strasuns-sociomantic** [sociomantic-tsunami/ocean#468]
Discussed with @scott-gibson-sociomantic recently - he had array of data with each element to be used to create some form of long-running background task. He wanted to run some small fixed amount of those (say, 5) at the same time and submit new one as soon as pool is not fully busy anymore.

Right now there is a way to await until all tasks in pool finish, but not until the first one finishes. Even more convenient would be to just have `task_pool.consume(array)` which would only work if pool has fixed size limit and execute new task each time limit is not reached until array gets empty.

**Comment by gavin-norman-sociomantic** [comment]
Interesting... This sounds really similar to the existing concept of a queue of tasks and a fixed pool of fibers on which they execute.

**Comment by mihails-strasuns-sociomantic** [comment]
In some way, though a bit less efficient. I would expect such feature to always be used with dedicated worker fiber pools though.

**Comment by gavin-norman-sociomantic** [comment]
What's the advantage of / reason for implementing something new, then?

**Comment by mihails-strasuns-sociomantic** [comment]
"long-running background task" is the key here - current system is not well-suited for mixing together tasks of a very different lifespan (because no matter what worker fiber count you pick, it may end up being either doing no work or overloaded). This is why dedicated pools had to be introduced in the first place.

**Comment by gavin-norman-sociomantic** [comment]
Right, this is what I mean. You'd have a dedicated fiber pool and queue just for this type of tasks. Does this require anything new to be implemented?
